iis-common/src/main/java/eu/dnetlib/iis/common/pig/udfs/DeduplicateIdsWithConfidence.java

```diff
@@ -13,55 +13,76 @@
+    
+    public DeduplicateIdsWithConfidence(int confidenceLevelPosition) {
+        this.confidenceLevelPosition = confidenceLevelPosition;
+    }
+    
+    public DeduplicateIdsWithConfidence(String confidenceLevelPosition) {
+        this(Integer.valueOf(confidenceLevelPosition));
+    }
+
+    //------------------------ PUBLIC --------------------------
+    
     @Override
     public DataBag exec(Tuple tuple) throws IOException {
         if (tuple == null || tuple.size() == 0) {
             return null;
         }
         DataBag db = (DataBag) tuple.get(0);
-        if (db==null) {
-        	return null;
+        if (db == null) {
+            return null;
         }
-        if (db.size()>1) {
-//        	deduplicating only if more than one element
-        	Iterator<Tuple> it = db.iterator();
-            Map<String,Tuple> dedupMap = new TreeMap<String, Tuple>(); 
+        if (db.size() > 1) {
+            // deduplicating only if more than one element
+            Iterator<Tuple> it = db.iterator();
+            Map<String, Tuple> dedupMap = new TreeMap<String, Tuple>();
             while (it.hasNext()) {
-            	Tuple currentTuple = it.next();
-            	Tuple storedTuple = dedupMap.get(currentTuple.get(0));
-            	if (storedTuple!=null) {
-            		if (currentTuple.get(1)!=null) {
-            			if (storedTuple.get(1)!=null) {
-//            				storing the one with higher confidence level
-            				if ((Float)currentTuple.get(1)>(Float)storedTuple.get(1)) {
-            					dedupMap.put((String) currentTuple.get(0), currentTuple);	
-            				}
-            			} else {
-            				dedupMap.put((String) currentTuple.get(0), currentTuple);
-            			}
-            		}
-            	} else {
-            		dedupMap.put((String) currentTuple.get(0), currentTuple);
-            	}
+                Tuple currentTuple = it.next();
+                Tuple storedTuple = dedupMap.get(currentTuple.get(0));
+                if (storedTuple != null) {
+                    Float currentTupleConfidenceLevel = (Float) currentTuple.get(this.confidenceLevelPosition);
+                    Float storedTupleConfidenceLevel = (Float) storedTuple.get(this.confidenceLevelPosition);
+                    if (currentTupleConfidenceLevel != null) {
+                        if (storedTupleConfidenceLevel != null) {
+                            // storing the one with higher confidence level
+                            if (currentTupleConfidenceLevel > storedTupleConfidenceLevel) {
+                                dedupMap.put((String) currentTuple.get(0), currentTuple);
+                            }
+                        } else {
+                            dedupMap.put((String) currentTuple.get(0), currentTuple);
+                        }
+                    }
+                } else {
+                    dedupMap.put((String) currentTuple.get(0), currentTuple);
+                }
             }
```

Comment:
I'd go with `updateStoredTupleWhenConfidenceLevelHigher(currentTuple, deduplicatedTuplesMap)` since when new candidate is not better than stored one we wouldn't need to perform any operation on map. I've also simplified some of the conditions.

As for the names - I fixed them where I could but generally we are dealing with tuples and we do not know what they are so it's difficult to name them appropriately. They could be anything with `identifier` as first field and `confidenceLevel` among other fields.